plugins/filter/lists.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.common.collections import is_sequence
+
+
+def remove_duplicates(lst):
+    seen = set()
+    result = []
+    for item in lst:
+        if item not in seen:
+            seen.add(item)
+            result.append(item)
```

Comment:
One way to implement this:
```suggestion
        try:
            if item not in seen:
                seen.add(item)
                result.append(item)
        except TypeError:
            # This happens for un-hashable values `item`. If this happens,
            # convert `seen` to a list and continue this way.
            seen = list(seen)
            if item not in seen:
                seen.add(item)
                result.append(item)
```